libcontainer/dmz/cloned_binary_linux.go

```diff
@@ -19,6 +20,23 @@ var (
 	_ SealFunc = sealFile
 )
 
+func isExecutable(f *os.File) bool {
+	if err := unix.Faccessat(int(f.Fd()), "", unix.X_OK, unix.AT_EACCESS|unix.AT_EMPTY_PATH); err == nil {
+		return true
+	} else if errors.Is(err, unix.EACCES) {
+		return false
+	}
+	path := "/proc/self/fd/" + strconv.Itoa(int(f.Fd()))
+	if err := unix.Access(path, unix.X_OK); err == nil {
```

Comment:
This is going to be a problem when CAP_DAC_OVERRIDE is set. Maybe it's better to use `unix.Faccessat(unix.AT_FDCWD, path, unix.X_OK, unix.AT_EACCESS)` or `exec.LookPath`.

For some context, see 8491d33482a4cfd37bad86b75a02c8f63464f427 and the links wherein.

Reply:
Or maybe it's not a problem since we proceed anyway. Still, makes sense to me to use AT_EACCESS.

Reply:
We already do an `AT_EACCESS` check. This is meant to be a fallback if `faccessat2` is unavailable (to be fair, I haven't checked how `golang.org/x/sys/unix` does the emulation of `AT_EACCESS`).

However, the goal here is to simply detect whether the filesystem has `noexec` in a race-free way -- `Access` should fail in that case. The issues with `CAP_DAC_OVERRIDE` that normally require `AT_EACCESS` isn't relevant to us here because the file was created by us and thus we definitely have `+x` permissions. In fact, I suspect just using `unix.Access` is sufficient, but I thought I might as well do `AT_EACCESS` as well.